interview-questions.md

```diff
@@ -6,62 +6,64 @@ First, without external resources. Challenge yourself to answer from memory as i
 
+I'm so sorry because I really don't know of these answers from the top of my head!!!!
+like i really don't know any of these off the top of my head
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: In my experience we use `this` to call on certain values inside of a react. To call on those values. There are differences when using react and javascript. When we call on we have to use `this`.
 
-  Researched answer:
+  Researched answer: Depending on how the keyword this is used it can refer to many things. One way that I have used recently is to refer to an object. When the keyword this is used in an object method, it refers to the object. I am not sure if I have used it any other way but if you use the keyword alone it refers to the global object. If you were to use it inside a function it refers to the global object as well.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a javascript library that has a user interface. I would use it because it is very useful. I know I use apps that use react such as facebook and netflix.
 
-  Researched answer:
+  Researched answer: React is a javascript library that creates an interactive UI. Companies like Facebook and Netflix use react because it is a scalable application. Using react can be extremely helpful when it comes down to cutting down development time. For example, react allows developers to create and use components that can be used interchangebly, basically making it reuseable. When something reloads it doesn't refresh the whole page rather it resets that small interaction and isolates it.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: I don't know the method by that name but I am sure that I have used it before. I know class components are normally followed by constructer(props) and super(props).
 
-  Researched answer:
+  Researched answer: After doing a little research I'm surprised to find out that the lifecycle method required for a react class component is render(). It seems that I may have missed it. When render is called you can get either of these items react elements, array and fragments, portals, strings, numbers, boolean, or null. I found out that render should not modify the component state.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: I know that you use JSX during react and you normally use it to "escape" a language. I know one of the difference is that you use a curly bracket as apart of its syntax.
 
-  Researched answer:
+  Researched answer: JSX stands for JavaScriptXml and fun fact it was created as a syntactic sugar for React. When you are writing in JSX you are writing in HTML and javascript together.
```

Comment:
For sure!  This can be super confusing at first since we have up to this point kept them separate, but this becomes really valuable as we scale into larger scare applications.  Another thing to note, JSX can only return one element which is why we wrap multiple tags inside react fragments.